GVFS/GVFS.Platform.POSIX/POSIXPlatform.cs

```diff
@@ -124,6 +124,30 @@ public override bool TryGetGVFSEnlistmentRoot(string directory, out string enlis
             return POSIXPlatform.TryGetGVFSEnlistmentRootImplementation(directory, out enlistmentRoot, out errorMessage);
         }
 
+        public override bool TryGetDefaultLocalCacheRoot(string enlistmentRoot, out string localCacheRoot, out string localCacheRootError)
+        {
+            string homeDirectory = Environment.GetEnvironmentVariable("HOME");
+            if (string.IsNullOrEmpty(homeDirectory))
+            {
+                localCacheRoot = null;
+                localCacheRootError = "Failed to read HOME environment variable";

```

Comment:
I think this message is correct when we catch the `SecurityException`, but is it correct if `HOME` is unset (for whatever reason). Should we distinguish between the two failure cases?

Reply:
> Should we distinguish between the two failure cases?

I think it's best to provide as much detail as we can in the error message, and so I will separate out the failure cases.